modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java

```diff
@@ -0,0 +1,250 @@
+public class EditArticleHeaderProductNavigationControlMenuEntryTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		LayoutTestUtil.addTypePortletLayout(_group);
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderInAnotherView() throws Exception {
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest(journalArticle)));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithGuestPermissions()
+		throws Exception {
+
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(journalArticle);
+
+		mockHttpServletRequest.addParameter(
+			"mvcRenderCommandName", "/journal/edit_article");
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithoutGuestPermissions()
+		throws Exception {
+
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		_resourcePermissionLocalService.deleteResourcePermissions(
+			_group.getCompanyId(), JournalArticle.class.getName(),
+			ResourceConstants.SCOPE_INDIVIDUAL,
+			journalArticle.getResourcePrimKey());
+
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(journalArticle);
+
+		mockHttpServletRequest.addParameter(
+			"mvcRenderCommandName", "/journal/edit_article");
+
+		Assert.assertTrue(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithoutJournalArticle()
+		throws Exception {
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest()));
+	}
+
+	private MockHttpServletRequest _getMockHttpServletRequest()
+		throws Exception {
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest(mockHttpServletRequest);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, mockHttpServletRequest);
+
+		MVCPortlet mvcPortlet = (MVCPortlet)_portlet;
+
+		mvcPortlet.render(
+			mockLiferayPortletRenderRequest,
+			new MockLiferayPortletRenderResponse());
```

Comment:
Why are you invoking the portlet? If you're doing it only to fill in request attributes, why not inject explicitly the display contexts in the test? For what I can see, those are the only things the control panel entry depends on. Making the whole portlet execute a render feels a bit overkill.

Reply:
Yes, I am doing that to fill the JournalDisplayContext that it is used in the `EditArticleHeaderProductNavigationControlMenuEntry`, What is your idea? . I wouldn't like to create the display context instance with reflection , I prefer to render the portlet and delegate to create things that the test need to execute it. I think that is more maintainable.

Reply:
Oh, I see. The display context is in another module. Makes sense then.